File: 2023/go/day01/part1.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
)

func part1() int{
	file, err := os.Open("input")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	total := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		nums := ""
		for _, c := range line {
			if '0' <= c && c <= '9' {
				nums += string(c)
			}

		}
		result := nums[:1] + nums[len(nums) - 1:]
		rasin, err := strconv.Atoi(result)
		if err != nil {
			log.Fatal(err)
		}
		total += rasin
	}
	return total
}

File: 2023/cpp/day01/src/part1.cpp
#include "part1.h"

#include <string>

int part1(std::ifstream& input){
	using namespace std;
	string line;
	int total = 0;

	while(getline(input, line)){
		int first = 0;
		int last = 0;
		for(int i = 0; i < line.length(); i++){
			if(line[i] >= '1' && '9' >= line[i] && first == 0){	
				first = int(line[i] - '0');
			}

			int ri = line.length() - i - 1;
			if(line[ri] >= '1' && '9' >= line[ri] && last == 0){	
				last = int(line[ri] - '0');	
			}
		}
		total += first * 10 + last;
	}
	return total;
}
